1.fee_bumping/fee_bumping.md

```diff
@@ -153,33 +153,224 @@ Bitcoin blockchain, such as exchanges or custodians:
+
+1. **DoS** - how do we prevent malicious actors from consuming node resources
+   by repeatedly broadcasting replacement transactions for zero cost?
+2. **Incentive alignment** - how do we make sure that replacement transactions
+   are incentive-compatible and miners want to mine them?
+
+Satoshi's first version of transaction replacement failed on both of these
+counts. The `nSequence` number could be incremented millions of times without
+changing the fee, wasting bandwidth and compute resources on all nodes in the
+network. The replacement logic also didn't require the replacement transaction
+to increase the fee, and could replace an old transaction with a new
+transaction even if the replacement had a lower fee. If a miner was running
+this software, the replacement behaviour would be costing him money since he'd
+be mining transactions with lower fees.
+
+For those reasons, Satoshi [disabled transaction replacement][disable replacement]
+in 2010. When asked later why he'd disabled that code, his response was:
+
+> Just to reduce surface area.  It wouldn't help with increasing tx fee. A tx
+> starts being valid at nLockTime.  It wouldn't work to have a tx that stops
+> being valid at a certain time; once a tx ever becomes valid, it must stay
+> valid permanently.
+
+Replacing a transaction does not invalidate the old transaction, so unless
+there is an incentive for the miner to use the new transaction, then just
+replacing an old transaction with one that has higher `nSequence` numbers
+is not incentive-compatible.
+
+## Replace-by-Fee
+
+Replace-by-fee is an effort to make transactions replaceable whilst avoiding
+the problems of exposing nodes to DoS vulnerabilities and misaligned
+incentives.  This is achieved by only allowing replacement if the replacement
+transaction has a higher fee than the replaced transaction. This prevents DoS
+attacks since every replacement transaction costs the attacker additional
+money, and ensures incentive alignment since the replacement transaction is
+financially advantageous to the miner.
+
+There are several [variants of replace-by-fee][rbf variants]. Although Bitcoin
+Core may implement *delayed RBF* at some point in the future (there is
+currently an [open PR][delayed rbf pr], which has support from several
+developers), currently almost all nodes on the network are thought to implement
+*opt-in RBF* as defined in [BIP 125][].
 
-TODO
-
-## Overview of what RBF is
-
-TODO
-
-## History of RBF
-
-TODO
 
 ## BIP 125
 
-TODO
-
-### 5 conditions for RBF
+BIP 125 is a specification for opt-in RBF written by Peter
+Todd in December 2015. It describes the conditions under which an unconfirmed
+transaction in a node's mempool will be replaced. BIP 125 is referred to as
+*opt-in* RBF since the node will only allow a transaction to be replaced if
+the sender has explicitly signaled that he wants the transaction to be
+replacable (or if the transaction is the descendant of an unconfirmed
```

Comment:
What's the part about the "descendant of an unconfirmed tx"? Do transactions spending outputs of rbf-txes automatically inherit replacability? I think this point should either be omitted or elaborated.